diffsync/store/redis.py

```diff
@@ -0,0 +1,242 @@
+"""RedisStore module."""
+import copy
+import uuid
+from pickle import loads, dumps  # nosec
+from typing import List, Mapping, Text, Type, Union, TYPE_CHECKING
+
+from redis import Redis
+from redis.exceptions import ConnectionError as RedisConnectionError
+from diffsync.exceptions import ObjectNotFound, ObjectStoreException, ObjectAlreadyExists
+from diffsync.store import BaseStore
+
+if TYPE_CHECKING:
+    from diffsync import DiffSyncModel
+
+REDIS_DIFFSYNC_ROOT_LABEL = "diffsync"
+
+
+class RedisStore(BaseStore):
+    """RedisStore class."""
+
+    def __init__(self, *args, store_id=None, host="localhost", port=6379, url=None, db=0, **kwargs):
+        """Init method for RedisStore."""
+        super().__init__(*args, **kwargs)
+
+        try:
+            if url:
+                self._store = Redis.from_url(url, db=db)
+            else:
+                self._store = Redis(host=host, port=port, db=db)
+
+            if not self._store.ping():
+                raise RedisConnectionError
+        except RedisConnectionError:
+            raise ObjectStoreException("Redis store is unavailable.") from RedisConnectionError
+
+        self._store_id = store_id if store_id else str(uuid.uuid4())[:8]
+
+        self._store_label = f"{REDIS_DIFFSYNC_ROOT_LABEL}:{self._store_id}"
+
+    def __str__(self):
+        """Render store name."""
+        return f"{self.name}({self._store_id})"
+
+    def get_all_model_names(self):
+        """Get all the model names stored.
+
+        Return:
+            List[str]: List of all the model names.
```

Comment:
Looking at the implementation here, I wonder if it would be more appropriate for `get_all_model_names` to return a Set rather than a List, since its members should be unique and order is not significant.